core/core-impl/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/generic/list/SubListChangeMoveSelectorConfig.java

```diff
@@ -154,6 +154,18 @@ public void visitReferencedClasses(Consumer<Class<?>> classVisitor) {
         }
     }
 
+    @Override
+    public boolean hasNearbySelectionConfig() {
+        return (subListSelectorConfig != null && ((subListSelectorConfig.getNearbySelectionConfig() != null)
```

Comment:
This condition is kinda insane.
Maybe simplify it by introducing boolean variables for its components?
The code will be longer, but people will actually be able to understand what the condition does.

(There are similar conditions elsewhere in this PR.)

Reply:
Let me try to simplify it.